httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CacheValidityPolicy.java

```diff
@@ -206,28 +207,62 @@ protected TimeValue getApparentAge(final HttpCacheEntry entry) {
         return TimeValue.ofSeconds(diff.getSeconds());
     }
 
+    /**
+     * Extracts and processes the Age value from an HttpCacheEntry by tokenizing the Age header value.
+     * The Age header value is interpreted as a sequence of tokens, and the first token is parsed into a number
+     * representing the age in delta-seconds. If the first token cannot be parsed into a number or if it exceeds
+     * Integer.MAX_VALUE, the Age value is set to MAX_AGE (in seconds).
+     * This method uses CacheSupport.parseTokens to robustly handle the Age header value.
+     * <p>
+     * Note: If the HttpCacheEntry contains multiple Age headers, only the first one is considered.
+     *
+     * @param entry The HttpCacheEntry from which to extract the Age value.
+     * @return The Age value in delta-seconds, or MAX_AGE in seconds if the Age value is invalid or exceeds Integer.MAX_VALUE.
+     * @throws NumberFormatException if the first token in the Age header value cannot be parsed into a number.
+     */
     protected long getAgeValue(final HttpCacheEntry entry) {
-        // This is a header value, we leave as-is
-        long ageValue = 0;
-        for (final Header hdr : entry.getHeaders(HttpHeaders.AGE)) {
-            long hdrAge;
+        final Header age = entry.getFirstHeader(HttpHeaders.AGE);
+        if (age != null) {
             try {
-                hdrAge = Long.parseLong(hdr.getValue());
-                if (hdrAge < 0) {
-                    hdrAge = MAX_AGE.toSeconds();
+                final AtomicLong ageValue = new AtomicLong(0);
```

Comment:
@arturobernalg You are still parsing all tokens instead of one (which is wasteful) and taking the last value from the sequence instead of the first. Please add test cases for various malformed values and make sure they all pass.

It is perfectly fine to use Long#parseLong. It is not going to be any less efficient.

Reply:
@ok2c 
Should be fine now.

Reply:
@arturobernalg Please use this code to extract the first token. It is massively simpler and is likely more efficient.
```
            final AtomicReference<String> firstToken = new AtomicReference<>();
            CacheSupport.parseTokens(age, token -> {
                firstToken.compareAndSet(null, token);
            });
            final String s = firstToken.get();
```
If you want to handle delta-seconds `"1" + Long.MAX_VALUE` please do it consistently for all headers, not just `Age`. For now I think accepting only valid long values that can be parsed with `Long#parseLong` should be good enough.

Reply:
@ok2c 
Changed